src/streaming/multiplex/mod.rs

```diff
@@ -17,11 +19,40 @@ pub use self::server::ServerProto;
 mod frame;
 pub use self::frame::Frame;
 
-
 pub mod advanced;
 
 /// Identifies a request / response thread
-pub type RequestId = u64;
+pub trait RequestId: Copy + Hash + Eq + Debug + 'static {}
+
+impl<T: Copy + Hash + Eq + Debug + 'static> RequestId for T {}
+
+/// `RequestIdSource` is used to generate at minimum session-wide unique identifiers of type `RId`.
+/// Uniqueness needs depend on the application and can be wider than single session.
+///
+/// Depending on the protocol the identifier can be generated or embedded in the message `T`.
+pub trait RequestIdSource<Id, T>: 'static {
```

Comment:
Currently implementing `tokio_proto::multiplex::ClientProto` requires user to also 

```
use tokio_proto::streaming::multiplex::RequestIdSource;
```

which seems quite ackward. Should this be moved or duplicated, or ...?